apps/organization/src/organization.service.ts

```diff
@@ -623,26 +623,38 @@ export class OrganizationService {
         userId
       );
 
-      let orgIds;
-      if (0 > getOrgs?.organizations?.length) {
-        orgIds = getOrgs?.organizations?.map(item => item.id);
+      if (0 === getOrgs?.organizations?.length) {
+        throw new NotFoundException(ResponseMessages.organisation.error.organizationNotFound);
       }
-       
-      const orgEcosystemDetails = await this._getOrgEcosystems(orgIds);
-      if (!orgEcosystemDetails || !Array.isArray(orgEcosystemDetails) || 0 === orgEcosystemDetails.length) {
-        throw new NotFoundException(ResponseMessages.ecosystem.error.ecosystemDetailsNotFound);
+
+      let orgIds;
+      let updatedOrgs;
+
+      if ('true' === process.env.IS_ECOSYSTEM_ENABLE) {
+        if (0 > getOrgs?.organizations?.length) {
+          orgIds = getOrgs?.organizations?.map(item => item.id);
+        }
+         
+        const orgEcosystemDetails = await this._getOrgEcosystems(orgIds);
+        if (!orgEcosystemDetails || !Array.isArray(orgEcosystemDetails) || 0 === orgEcosystemDetails.length) {
+          throw new NotFoundException(ResponseMessages.ecosystem.error.ecosystemDetailsNotFound);
+        }
+    
+        updatedOrgs = getOrgs.organizations.map(org => {
+          const matchingEcosystems = orgEcosystemDetails
+            .filter(ecosystem => ecosystem.orgId === org.id)
+            .map(ecosystem => ({ ecosystemId: ecosystem.ecosystemId }));
+          return {
+            ...org,
+            ecosystemOrgs: 0 < matchingEcosystems.length ? matchingEcosystems : []
+          };
+        });
+      } else {
```

Comment:
This `else` is for org without any ecosystem associated, right?

Reply:
Yes, Correct